src/viam/sdk/module/service.cpp

```diff
@@ -250,7 +250,14 @@ void ModuleService::serve() {
 }
 
 ModuleService::~ModuleService() {
-    // TODO(RSDK-5509): Run registered cleanup functions here.
+    try {
+        cleanup_function_();
+    } catch (const std::exception& xcp) {
+        BOOST_LOG_TRIVIAL(error) << "Module cleanup function failed with exception " << xcp.what();
+    } catch (...) {
+        BOOST_LOG_TRIVIAL(error) << "Module cleanup function failed with unknown exception";
+    }
```

Comment:
This double catch is me trying to ensure that `server_` is always shut down and `parent_` is always closed (unless `cleanup_function_` segfaults or blocks forever or something).

Reply:
Even more than that, destructors are default `noexcept` so throwing from the cleanup function would lead to an immediate call to `std::terminate` if you didn't have this.